transfermethodui/src/test/java/com/hyperwallet/android/ui/transfermethod/view/widget/AbstractMaskedInputWidgetTest.java

```diff
@@ -22,8 +27,11 @@ public AbstractMaskedInputWidgetTest() {
     }
 
     @Test
-    public void testFormatToDisplay_usingExcelData() {
-        // TODO for Mani
+    @FileParameters("src/test/resources/formatter_mask_test_scenarios.csv")
+    public void testFormatToDisplay_usingExcelData(String description, String pattern, String scenario,
+            String inputValue, String formattedValue) {
+        assertThat("Description: " + description + "; Scenario: " + scenario,
```

Comment:
I think it would be useful to add the pattern too.
<img width="880" alt="Screen Shot 2019-12-04 at 4 10 46 PM" src="https://user-images.githubusercontent.com/28124536/70192505-c433b680-16b0-11ea-9b49-ccb2d3fabcab.png">
<img width="592" alt="Screen Shot 2019-12-04 at 4 11 03 PM" src="https://user-images.githubusercontent.com/28124536/70192506-c433b680-16b0-11ea-84df-009712f22a11.png">

Reply:
Yeah as I said in Jira, some scenarios are not correct in the spreadsheet. The failures tend to contradict the actual outputs.

After reviewing, I think it would be better to do a parameterized test.  See the samples I created below.
As well, reading from file seemed to alter the value of how the backslash character is interpreted when compared to the Android App consuming the graphql response.

Such as `##\\####`, when debugging the true value:
In file read, it becomes: ##\\####
But normal Java String it becomes: ##\####

Reply:
added @fmattos-hw Thanks.
Yes i got @dnip-hw. So i updated existing test case. Also added new test case based on your idea. So you can choose any one from those two test cases(**testFormatToDisplay_usingExcelData**, **testFormatToDisplay_fromExcelData**). Because i implemented to reduce the number of lines. 
FYI, I also got some failed test cases after code update. Most of the failure is ending with special character for input value shorter length than the pattern.
<img width="1393" alt="Screenshot 2019-12-05 at 8 26 06 PM" src="https://user-images.githubusercontent.com/52773997/70246627-0981e200-179e-11ea-956d-ecdc2d2eff03.png">
<img width="1437" alt="Screenshot 2019-12-05 at 8 26 15 PM" src="https://user-images.githubusercontent.com/52773997/70246629-0981e200-179e-11ea-8f58-4eded5e8b9c9.png">
<img width="1443" alt="Screenshot 2019-12-05 at 8 26 26 PM" src="https://user-images.githubusercontent.com/52773997/70246632-0a1a7880-179e-11ea-9315-4f70e170360e.png">
<img width="1441" alt="Screenshot 2019-12-05 at 8 26 34 PM" src="https://user-images.githubusercontent.com/52773997/70246633-0a1a7880-179e-11ea-9f98-330695830d68.png">
<img width="1443" alt="Screenshot 2019-12-05 at 8 26 42 PM" src="https://user-images.githubusercontent.com/52773997/70246634-0a1a7880-179e-11ea-874f-bcdf4d90796c.png">
<img width="1441" alt="Screenshot 2019-12-05 at 8 26 50 PM" src="https://user-images.githubusercontent.com/52773997/70246635-0ab30f00-179e-11ea-8a6c-f1823f408a67.png">
<img width="1440" alt="Screenshot 2019-12-05 at 8 26 59 PM" src="https://user-images.githubusercontent.com/52773997/70246636-0b4ba580-179e-11ea-87c2-2253ed55c959.png">
<img width="1444" alt="Screenshot 2019-12-05 at 8 27 09 PM" src="https://user-images.githubusercontent.com/52773997/70246639-0b4ba580-179e-11ea-9f06-b3971d853487.png">
<img width="1443" alt="Screenshot 2019-12-05 at 8 27 17 PM" src="https://user-images.githubusercontent.com/52773997/70246643-0be43c00-179e-11ea-88d2-5ac61f0d6712.png">

Reply:
I checked all 9 failing cases, I would argue that the expected output from Excel is wrong. I corrected them to the expected value. I list them below and put it on the ticket:
```
// fail 1, row 22, excel is wrong (11#), shouldn't add suffix
// fail 2, row 47, excel is wrong (aa@), shouldn't add suffix
// fail 3, row 67, excel is wrong (11), user entered matching dash, we should enter it too
// fail 4, row 73, excel is wrong (11*), shouldn't add suffix
// fail 5, row 76, excel is wrong (11*-N), * character only accepts 0-9AZaz
// fail 6, row 77, excel is wrong (aa*-a), * character only accepts 0-9AZaz
// fail 7, row 87, excel is wrong (1a-), * character only accepts 0-9AZaz
// fail 8, row 99, excel is wrong (@), shouldn't add prefix if no valid characers inputted
// fail 9, row 100, excel is wrong (@a1a#a1a*), shouldn't add suffix```